arenadata/readme.md

```diff
@@ -108,6 +108,17 @@ bash arenadata/scripts/run_behave_tests.bash
 bash arenadata/scripts/run_behave_tests.bash gpstart gpstop
 ```
 
+## Building with alternative compiler
+
+You can specify a custom compiler to build gpdb with. For that, you need to specify CC and CXX arguments
+with --build-arg option. You can also run syntax check only with USE_CLANG_ANALYZER and skip unit tests
+with SKIP_UNITTESTS flag. For example, checking syntax only with clang without running unit tests requires:
+```bash
+docker build -t gpdb7_regress:latest -f arenadata/Dockerfile . --build-arg CC=clang --build-arg CXX=clang++ --build-arg USE_CLANG_ANALYZER=true --build-arg SKIP_UNITTESTS=true
```

Comment:
The unit tests are not skipped and not passed
```mocking /home/gpadmin/gpdb_src/src/backend/utils/time/snapmgr.c
LOG:  FTS: cannot establish libpq connection to (content=1, dbid=4): 
LOG:  FTS: failed to read from (content=0, dbid=2) state=0, retry_count=0, conn->asyncStatus=0 
LOG:  FTS: error response from (content=0, dbid=2) state=2, retry_count=0, conn->asyncStatus=0 
WARNING:  (01000) ERROR: FTS double fault detected (content=0) primary dbid=2, mirror dbid=3
WARNING:  (01000) ERROR: FTS double fault detected (content=0) primary dbid=2, mirror dbid=3
LOG:  FTS timeout detected for (content=0, dbid=2) state=0, retry_count=0, timeout_count=0 
[          OK ] test_ftsConnect_FTS_PROBE_SEGMENT
[          OK ] test_ftsConnect_one_failure_one_success
[          OK ] test_ftsConnect_ftsPoll
[          OK ] test_ftsSend_success
[          OK ] test_ftsReceive_success
[          OK ] test_ftsReceive_when_fts_handler_FATAL
[          OK ] test_ftsReceive_when_fts_handler_ERROR
[          OK ] test_processRetry_wait_before_retry
[          OK ] test_processResponse_for_zero_segment
[          OK ] test_processResponse_for_FtsIsActive_false
[          OK ] test_processResponse_multiple_segments
[          OK ] test_PrimayUpMirrorUpSync_to_PrimaryUpMirrorUpNotInSync
[          OK ] test_PrimayUpMirrorUpSync_to_PrimaryUpMirrorDownNotInSync
[          OK ] test_PrimayUpMirrorUpSync_to_PrimaryDown_to_MirrorPromote
[          OK ] test_PrimayUpMirrorUpNotInSync_to_PrimayUpMirrorUpSync
[          OK ] test_PrimayUpMirrorUpNotInSync_to_PrimayUpMirrorUpNotInSync
[          OK ] test_PrimayUpMirrorUpNotInSync_to_PrimaryUpMirrorDownNotInSync
[          OK ] test_PrimayUpMirrorUpNotInSync_to_PrimaryDown
[          OK ] test_PrimaryUpMirrorDownNotInSync_to_PrimayUpMirrorUpSync
[          OK ] test_PrimaryUpMirrorDownNotInSync_to_PrimayUpMirrorUpNotInSync
[          OK ] test_PrimaryUpMirrorDownNotInSync_to_PrimayUpMirrorDownNotInSync
[          OK ] test_PrimaryUpMirrorDownNotInSync_to_PrimaryDown
[          OK ] test_probeTimeout
[          OK ] test_FtsWalRepInitProbeContext_initial_state
[=============] 24 tests ran
[ PASSED      ] 24 tests
make[4]: *** [../../../../../src/Makefile.mock:25: cdbsenddummypacket-check] Aborted
make[3]: *** [../../../../src/backend/common.mk:50: unittest-check-local] Error 2
make[2]: *** [../../../src/backend/common.mk:47: unittest-check-motion-recurse] Error 2
make[1]: *** [common.mk:47: unittest-check-cdb-recurse] Error 2
make: *** [GNUmakefile:233: unittest-check] Error 2
The command '/bin/sh -c bash /home/gpadmin/gpdb_src/concourse/scripts/compile_gpdb.bash' returned a non-zero code: 2
```

Reply:
> unit tests are not skipped

I think it's because `ARG` is not `ENV`.

Reply:
Fixed